### PR TITLE
added gpg2 as fallback

### LINKED ISSUE #1796: rpm_key: Fedora 22: Failed to find required executable gpg
Fedora 22 has no gpg executable but only gpg2.

Is it my responsibility to provide a link gpg -> gpg2 or should the module cover this out of the box?

### LINKED ISSUE #1797: setup on Fedora 22: ansible_pkg_mgr should be 'dnf'
Issue Type: Bug Report
Ansible Version: 1.9.2
Summary:

setup module should return dnf (instead of yum) for ansible_pkg_mgr on Fedora 22

Expected Results:
"ansible_pkg_mgr": "dnf"

Actual Results:
"ansible_pkg_mgr": "yum"

### PR DESCRIPTION
**Pull request by bcoca**
fixes #1796

### CODE REVIEW

**Comment by gregdek** [comment]
Assuming that @bcoca is waiting to merge this for some opportune moment. If you need review, lemme know and we can put it into whatever state you'd like. :)

**Comment by gregdek** [comment]
Actually, I just realized that @bcoca is not the author here!

@hacosta Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by bcoca** [comment]
@gregdek yes i am

File: packaging/os/rpm_key.py [hunk 1]
```diff
@@ -141,7 +141,14 @@ def normalize_keyid(self, keyid):
...
 
     def getkeyid(self, keyfile):
-        gpg = self.module.get_bin_path('gpg', True)
+
+        gpg = self.module.get_bin_path('gpg')
+        if not gpg:
+            gpg = self.module.get_bin_path('gpg2')
+
+        if not gpg:
+            self.json_fail(msg="rpm_key requires a command lne gpg or gpg2, none found")
```
Comment (alxgu) [inline packaging/os/rpm_key.py]: typo lne => line

**Pull request merged**